types/presto-client/index.d.ts

```diff
@@ -200,7 +200,18 @@ export interface RuntimeStats {
     wallTimeMillis: number;
     cpuTimeMillis: number;
     userTimeMillis: number;
-    state: "QUEUED" | "PLANNING" | "STARTING" | "RUNNING" | "FINISHED" | "CANCELED" | "FAILED";
+    state:
+        | "WAITING_FOR_PREREQUISITES"
+        | "QUEUED"
+        | "WAITING_FOR_RESOURCES"
+        | "DISPATCHING"
+        | "PLANNING"
+        | "STARTING"
+        | "RUNNING"
+        | "FINISHING"
+        | "FINISHED"
+        | "CANCELED"
```

Comment:
To confirm, `CANCELED` is left here for historical reasons as before https://github.com/prestodb/presto/commit/4bf8df1484935a000619377dab6ed35778fff2ad it was a valid state, but since then, it should not be possible to see this (and would now see `FAILED` with an error message to indicate it was canceled)?

Reply:
yes, I personally observe `CANCELED` in my system; I don't know if it was every fully removed from servers or if I'm interacting with older ones; I think it is safe/best to leave it it for legacy support

Reply:
Could you add a comment indicating that this is not in `QueryState` and is kept for legacy reasons?

Reply:
ah, yes, will do

Reply:
```suggestion
        | "CANCELED" // legacy value, see https://github.com/prestodb/presto/commit/4bf8df1484935a000619377dab6ed35778fff2ad#diff-0e3098a2bd5426fc1a01f56708912cedcdf7ce5d2e0f7adcdb70b93080c8733a
```

Reply:
done